src/app/subscriptions.ts

```diff
@@ -17,69 +17,35 @@ import { AppStore } from "./state/store";
 import { rewardSlice } from "./state/rewardSlice";
 
 export type RDT = ReturnType<typeof RadixDappToolkit>;
-
-let rdtInstance: null | RDT = null;
-export function getRdt() {
-  return rdtInstance;
-}
-
-export function getRdtOrThrow() {
-  const rdt = getRdt();
-  if (!rdt) {
-    throw new Error("RDT initialization failed");
-  }
-  return rdt;
-}
-function setRdt(rdt: RDT) {
-  rdtInstance = rdt;
-}
+export let rdt: RDT;
```

Comment:
I'm surprised TS allows you to do that without an error. You are declaring `rdt` as RDT type always, but not assigning a value immediately. That means at some point in time during the application init this variable is `undefined` and we need to not forget to deal with such cases accordingly everywhere we are using the rdt instance. 

```suggestion
export let rdt: RDT | undefined;
```

Reply:
@fliebenberg what about this one? Are you sure rdt will always be initialized before usage? Even on slow CPU clients?

Reply:
I'm thinking it might be good idea that we can trust rdt to be always initialized, otherwise my opinion is that the app should fail as it's an important dependency for many different slices in the application. Also, as far as I know we wouldn't want control of initializing rdt to fall on random slices.

Reply:
GPT4 suggest adding `--strictNullChecks`:

"TypeScript allows this behavior because the variable's type (RDT) does not exclude undefined or null by default. If you want TypeScript to enforce that a variable always has a value of its specific type from the moment it is defined, you can configure TypeScript to use strict null checks (--strictNullChecks option)"

Reply:
I agree that it should be fine for the app to crash (essentially it wont start) if rdt is not initialised. rdt is initialised with the RadixDappToolkit function which is a synchronous function that returns an instance of the RadixDappToolkit type. So it will be completely initialized or fail (e.g. if some of the settings in the code is incorrect) before the code can go on to load the rest of the app.

Reply:
We probably need some global state variable that watches if the app is initialized and only render all the child components when the init script has finished.

Reply:
The best would be if we did not have to run the initSubscriptions in an effect at all. It is not an effect, it is just normal synchronous code that needs to run once when the app starts. Unfortunately I dont know Next well enough to know where that is.

Reply:
When I try to move the initSubscriptions function out of the useEffect I get this build error:

Error occurred prerendering page "/_not-found". Read more: https://nextjs.org/docs/messages/prerender-error
ReferenceError: document is not defined
    at oz (C:\Fred\Coding\dexter\website\.next\server\chunks\511.js:52:79749)
    at lA (C:\Fred\Coding\dexter\website\.next\server\chunks\511.js:52:165524)
    at x (C:\Fred\Coding\dexter\website\.next\server\chunks\361.js:2:13269)
    at V (C:\Fred\Coding\dexter\website\.next\server\chunks\361.js:1:8863)
    at nT (C:\Fred\Coding\dexter\website\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:12:46138)
    at nT (C:\Fred\Coding\dexter\website\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:12:59937)
    at nO (C:\Fred\Coding\dexter\website\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:12:62795)
    at nI (C:\Fred\Coding\dexter\website\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:12:65445)
    at nj (C:\Fred\Coding\dexter\website\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:12:64852)
    at nO (C:\Fred\Coding\dexter\website\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:12:63041)

Reply:
This error happens because you need to make sure the init runs only in browser (rdt doesn't work otherwise), in react `useEffect` runs only in browser, maybe you can find some other way, but I think having a state variable is the simplest: 
- create a local variable with useState in layout component (something like isInitilized), set it to false
- after init finishes successfully set this variable to true
- render children only when isInitilized is true